plugins/USBPrinting/USBPrinterOutputDevice.py

```diff
@@ -566,7 +566,7 @@ def _sendNextGcodeLine(self):
             if line == "M0" or line == "M1":
                 line = "M105"  # Don't send the M0 or M1 to the machine, as M0 and M1 are handled as an LCD menu pause.
             if ("G0" in line or "G1" in line) and "Z" in line:
-                z = float(re.search("Z([0-9\.]*)", line).group(1))
+                z = float(re.search("Z([-0-9\.]*)", line).group(1))
```

Comment:
If the engine produces invalid g-code such as `G1 Z-0.6-9.2`or something, the proposed solution would give a ValueError that it can't convert this gibberish to a float. The more strict regex that you propose would give a bounds error because group 1 doesn't exist. I think the former is more appropriate, though it might not be as elegant.